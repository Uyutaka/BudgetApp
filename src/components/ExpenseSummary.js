import React from 'react';
import {connect } from 'react-redux';
import numeral from 'numeral';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';

export const ExpenseSummary = ({expenseCount, expenseTotal}) =>{
    const expenseWord = expenseCount === 1 ? 'expense': 'expenses';
    const formattedExpensesTotala = numeral(expenseTotal/100).format('$0,0.00');

    return (
        <div>
            <h1>Viewing {expenseCount} {expenseWord} totaling {formattedExpensesTotala}</h1>
        </div>

    );
}

const mapStateToProps = (state) => {
    const visibleExpense = selectExpenses(state.expenses, state.filters);
    return {
        expenseCount: visibleExpense.length,
        expenseTotal: selectExpensesTotal((visibleExpense))
    };
};

export default connect(mapStateToProps)(ExpenseSummary);