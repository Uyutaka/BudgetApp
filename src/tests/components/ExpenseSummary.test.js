import React from 'react';
import {shallow} from 'enzyme';
import {ExpenseSummary} from "../../components/ExpenseSummary";

test('should correctly render ExpensesSummary with 1 expense', ()=>{
    const wrapper = shallow(<ExpenseSummary expenseCount={1} expenseTotal={324}/>)
    expect(wrapper).toMatchSnapshot();
});

test('should correctly render ExpensesSummary with multiple expenses', ()=>{
    const wrapper = shallow(<ExpenseSummary expenseCount={3} expenseTotal={3243768427}/>)
    expect(wrapper).toMatchSnapshot();
});