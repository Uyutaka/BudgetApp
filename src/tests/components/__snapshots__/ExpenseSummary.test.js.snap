// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render ExpensesSummary with 1 expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totaling 
    $3.24
  </h1>
</div>
`;

exports[`should correctly render ExpensesSummary with multiple expenses 1`] = `
<div>
  <h1>
    Viewing 
    3
     
    expenses
     totaling 
    $32,437,684.27
  </h1>
</div>
`;
